### ISSUE TITLE
Kotlin POJO generator produces wrong equals/hashCode for binary types

### ISSUE THREAD

**Issue by svilenivanovsmule** [jOOQ/jOOQ#11016]
### Expected behavior
This is a follow up to #10805. When comparing arrays in Kotlin one must use `.contentEquals(...)` and `contentHashCode(...)` to make generated Kotlin POJOs semantically identical to Java POJOs

### Actual behavior
JOOQ uses the default `hashCode()` when it comes to array properties which does not take into account hash code of the elements in the array. Similarly, it produces `==` operator in the `equals()` method. The result is also highlighted by IntelliJ IDEA:

![image](https://user-images.githubusercontent.com/1734398/100063849-73c1e580-2e3a-11eb-9402-562cc3a762fa.png)

### Steps to reproduce the problem

Switch to `KotlinGenerator` and enable `pojosEqualsAndHashCode`

### Versions

- jOOQ: 3.14.3
- Java: 11
- Database (include vendor): MySQL 8
- OS: Linux
- JDBC Driver (include name if inofficial driver): ConnectorJ/8.0.22

**Comment by lukaseder** [comment]
Thanks. We're currently generating `java.util.Arrays.equals()` and `java.util.Arrays.hashCode()` calls. What am I missing?

**Comment by lukaseder** [comment]
Hmm, MySQL doesn't support arrays. Is this related to some `Converter` usage? Can you please provide a complete reproducer (e.g. based on the suggested template: https://github.com/jOOQ/jOOQ-mcve)

**Comment by svilenivanovsmule** [comment]
MySQL BLOBs are treated as byte[] (Java) / ByteArray (Kotlin). The reproducible scenario is the same as in #10805. 

Database schema:

```sql
CREATE TABLE `testBlob` (
  `blobCol` blob
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci 
```

Configuration:
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<configuration xmlns="http://www.jooq.org/xsd/jooq-codegen-3.14.0.xsd">
    <onError>FAIL</onError>
    <jdbc>
        <driver>com.mysql.cj.jdbc.Driver</driver>
        <url>jdbc:mysql://localhost:3306</url>
        <username>root</username>
        <password>xxxxx</password>
    </jdbc>
    <generator>
        <name>org.jooq.codegen.KotlinGenerator</name>
        <database>
            <name>org.jooq.meta.mysql.MySQLDatabase</name>
            <regexMatchesPartialQualification>true</regexMatchesPartialQualification>
            <sqlMatchesPartialQualification>true</sqlMatchesPartialQualification>
            <includes>.*</includes>
            <excludes></excludes>
            <includeExcludeColumns>false</includeExcludeColumns>
            <includeTables>true</includeTables>
            <includeEmbeddables>true</includeEmbeddables>
            <includeRoutines>true</includeRoutines>
            <includeTriggerRoutines>false</includeTriggerRoutines>
            <includePackages>true</includePackages>
            <includePackageRoutines>true</includePackageRoutines>
            <includePackageUDTs>true</includePackageUDTs>
            <includePackageConstants>true</includePackageConstants>
            <includeUDTs>true</includeUDTs>
            <includeDomains>true</includeDomains>
            <includeSequences>true</includeSequences>
            <includeIndexes>true</includeIndexes>
            <includePrimaryKeys>true</includePrimaryKeys>
            <includeUniqueKeys>true</includeUniqueKeys>
            <includeForeignKeys>true</includeForeignKeys>
            <includeCheckConstraints>true</includeCheckConstraints>
            <includeSystemIndexes>false</includeSystemIndexes>
            <includeSystemCheckConstraints>false</includeSystemCheckConstraints>
            <includeInvisibleColumns>true</includeInvisibleColumns>
            <recordVersionFields></recordVersionFields>
            <recordTimestampFields></recordTimestampFields>
            <syntheticIdentities></syntheticIdentities>
            <syntheticPrimaryKeys></syntheticPrimaryKeys>
            <overridePrimaryKeys></overridePrimaryKeys>
            <dateAsTimestamp>false</dateAsTimestamp>
            <ignoreProcedureReturnValues>false</ignoreProcedureReturnValues>
            <unsignedTypes>true</unsignedTypes>
            <integerDisplayWidths>true</integerDisplayWidths>
            <inputCatalog></inputCatalog>
            <outputCatalogToDefault>false</outputCatalogToDefault>
            <inputSchema>test</inputSchema>
            <outputSchemaToDefault>false</outputSchemaToDefault>
            <schemaVersionProvider></schemaVersionProvider>
            <catalogVersionProvider></catalogVersionProvider>
            <orderProvider></orderProvider>
            <forceIntegerTypesOnZeroScaleDecimals>true</forceIntegerTypesOnZeroScaleDecimals>
            <logSlowQueriesAfterSeconds>5</logSlowQueriesAfterSeconds>
            <logSlowResultsAfterSeconds>5</logSlowResultsAfterSeconds>
        </database>
        <generate>
            <indexes>true</indexes>
            <relations>true</relations>
            <sequenceFlags>true</sequenceFlags>
            <implicitJoinPathsToOne>true</implicitJoinPathsToOne>
            <deprecated>true</deprecated>
            <deprecationOnUnknownTypes>true</deprecationOnUnknownTypes>
            <instanceFields>true</instanceFields>
            <generatedAnnotation>false</generatedAnnotation>
            <generatedAnnotationType>DETECT_FROM_JDK</generatedAnnotationType>
            <generatedAnnotationDate>false</generatedAnnotationDate>
            <nonnullAnnotation>true</nonnullAnnotation>
            <nonnullAnnotationType>javax.annotation.Nonnull</nonnullAnnotationType>
            <nullableAnnotation>true</nullableAnnotation>
            <nullableAnnotationType>javax.annotation.Nullable</nullableAnnotationType>
            <constructorPropertiesAnnotation>false</constructorPropertiesAnnotation>
            <routines>true</routines>
            <sequences>true</sequences>
            <udts>true</udts>
            <queues>true</queues>
            <links>true</links>
            <keys>true</keys>
            <tables>true</tables>
            <embeddables>true</embeddables>
            <records>true</records>
            <recordsImplementingRecordN>true</recordsImplementingRecordN>
            <pojos>false</pojos>
            <pojosEqualsAndHashCode>true</pojosEqualsAndHashCode>
            <pojosToString>true</pojosToString>
            <pojosAsJavaRecordClasses>false</pojosAsJavaRecordClasses>
            <pojosAsScalaCaseClasses>true</pojosAsScalaCaseClasses>
            <pojosAsKotlinDataClasses>true</pojosAsKotlinDataClasses>
            <immutablePojos>true</immutablePojos>
            <serializablePojos>true</serializablePojos>
            <interfaces>false</interfaces>
            <immutableInterfaces>false</immutableInterfaces>
            <serializableInterfaces>true</serializableInterfaces>
            <daos>false</daos>
            <jpaAnnotations>false</jpaAnnotations>
            <validationAnnotations>false</validationAnnotations>
            <springAnnotations>false</springAnnotations>
            <globalObjectReferences>true</globalObjectReferences>
            <globalCatalogReferences>true</globalCatalogReferences>
            <globalSchemaReferences>true</globalSchemaReferences>
            <globalDomainReferences>true</globalDomainReferences>
            <globalTableReferences>true</globalTableReferences>
            <globalSequenceReferences>true</globalSequenceReferences>
            <globalUDTReferences>true</globalUDTReferences>
            <globalRoutineReferences>true</globalRoutineReferences>
            <globalQueueReferences>true</globalQueueReferences>
            <globalLinkReferences>true</globalLinkReferences>
            <globalKeyReferences>true</globalKeyReferences>
            <globalIndexReferences>true</globalIndexReferences>
            <javadoc>true</javadoc>
            <comments>true</comments>
            <commentsOnCatalogs>true</commentsOnCatalogs>
            <commentsOnSchemas>true</commentsOnSchemas>
            <commentsOnTables>true</commentsOnTables>
            <commentsOnColumns>true</commentsOnColumns>
            <commentsOnEmbeddables>true</commentsOnEmbeddables>
            <commentsOnUDTs>true</commentsOnUDTs>
            <commentsOnAttributes>true</commentsOnAttributes>
            <commentsOnPackages>true</commentsOnPackages>
            <commentsOnRoutines>true</commentsOnRoutines>
            <commentsOnParameters>true</commentsOnParameters>
            <commentsOnSequences>true</commentsOnSequences>
            <commentsOnLinks>true</commentsOnLinks>
            <commentsOnQueues>true</commentsOnQueues>
            <commentsOnKeys>true</commentsOnKeys>
            <sources>true</sources>
            <sourcesOnViews>true</sourcesOnViews>
            <fluentSetters>false</fluentSetters>
            <javaBeansGettersAndSetters>false</javaBeansGettersAndSetters>
            <varargSetters>false</varargSetters>
            <generatedSerialVersionUID>CONSTANT</generatedSerialVersionUID>
            <maxMembersPerInitialiser>500</maxMembersPerInitialiser>
            <fullyQualifiedTypes></fullyQualifiedTypes>
            <emptyCatalogs>false</emptyCatalogs>
            <emptySchemas>false</emptySchemas>
            <javaTimeTypes>true</javaTimeTypes>
            <newline>\n</newline>
        </generate>
        <target>
            <packageName>com.example</packageName>
            <directory>jooq/build/generated/sources/jooq</directory>
            <encoding>UTF-8</encoding>
            <clean>true</clean>
        </target>
    </generator>
</configuration>
```

**Comment by svilenivanovsmule** [comment]
> Thanks. We're currently generating java.util.Arrays.equals() and java.util.Arrays.hashCode() calls. What am I missing?

That's right - this is for Java POJOs. The issues exhibits itself in Kotlin POJOs

**Comment by lukaseder** [comment]
> > Thanks. We're currently generating java.util.Arrays.equals() and java.util.Arrays.hashCode() calls. What am I missing?
> 
> That's right - this is for Java POJOs. The issues exhibits itself in Kotlin POJOs

What I meant is we're generating these calls for array types in Kotlin as well, but perhaps not for blobs. I'll check that.

**Comment by lukaseder** [comment]
Indeed, we're not doing that for binary types yet

**Comment by lukaseder** [comment]
Fixed for 3.15.0 and 3.14.4 (#11018)
